config/opal_check_os_flavors.m4

```diff
@@ -22,7 +22,7 @@ AC_DEFUN([OPAL_CHECK_OS_FLAVOR_SPECIFIC],
     AC_MSG_CHECKING([$1])
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
      [[#ifndef $1
-      error: this is not $1
+      #error: this is not $1
```

Comment:
I don't understand why this change is the right change.  For reasons documented at https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.71/html_node/C-Compiler.html, using #error in an autoconf test is a no-no.  THe autoconf tests themselves use "choke me" instead of our nice pretty print, but I have difficulty seeing how ours can cause more problems than what AC does in general.

Reply:
Thanks, I didn't realise that was the autoconf guidance.

Let me explain then you can tell me what you think the best approach is:
- In general, we're looking for new Clang 16 (and GCC 14) breakage which is painful if it's in config.log
- Easiest way is grepping for -Wimplicit-int, -Wimplicit-function-declaration, etc
- The current state of the macro leads to a bunch of false positives:
```
conftest.c:84:7: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
      error: this is not __OpenBSD__
      ^
      int
conftest.c:84:12: error: expected ';' after top level declarator
      error: this is not __OpenBSD__
           ^
           ;
conftest.c:84:14: error: unknown type name 'this'
      error: this is not __OpenBSD__
             ^
conftest.c:84:21: error: expected ';' after top level declarator
      error: this is not __OpenBSD__
                    ^
                    ;
4 errors generated.
configure:12189: $? = 1
```

This isn't a common occurrence and it's the first time so far in the general porting effort that I've come across this construct.

If you insist on keeping it as it is, that's fine, it just makes life a lot easier when going through literally 1000s of these packages and looking for problems. It'll save someone else looking at it and realising it's actually fine in this case.

What do you think?

Reply:
Why are you not tripping over the same problem for the "Choke me" that is in every configure script out there that checks for GNU compiler and therefore runs this code:

```
int
main ()
{
#ifndef __GNUC__
       choke me
#endif

  ;
  return 0
```

Reply:
I think trying to grep that out of config.log is a path to madness.  But I have no objection to a patch that fits with Autoconf's guidance.  Not following Autoconf's guidance there seems like you're going to be generally chasing your tail with Autoconf macros themselves.

Reply:
There's not really an alternative for what we're doing (see https://lwn.net/Articles/913505/). Surprisingly the autoconf macros themselves have been mostly fine, although they're going to make a new release for a few nits shortly. But I won't bore you with the details.

How's the latest thing I pushed?